src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -94,22 +94,28 @@
   uses. JARS_NO_REQUIRE was the legacy way to turn off jar loading but is
   being phased out in favor of JARS_REQUIRE.  As of JRuby 1.7.20, only
   JARS_NO_REQUIRE is honored.  Setting both of those here for forward
-  compatibility."
+  compatibility.
+
+  We also merge an environment-vars map with the config to allow for whitelisted
+  environment variables to be visible to the Ruby code. This map is by default
+  set to {} if the user does not specify it in the configuration file."
   [env :- jruby-schemas/EnvMap
-   gem-home :- schema/Str]
+   gem-home :- schema/Str
+   config :- (schema/pred map?)]
   (let [whitelist ["HOME" "PATH"]
         clean-env (select-keys env whitelist)]
-    (assoc clean-env
-      "GEM_HOME" gem-home
-      "JARS_NO_REQUIRE" "true"
-      "JARS_REQUIRE" "false")))
+    (merge (assoc clean-env
+                  "GEM_HOME" gem-home
+                  "JARS_NO_REQUIRE" "true"
+                  "JARS_REQUIRE" "false")
+           (:environment-vars config))))
 
 (schema/defn ^:always-validate default-initialize-scripting-container :- jruby-schemas/ConfigurableJRuby
   "Default lifecycle fn for initializing the settings on the scripting
   container. Currently it just sets the environment variables."
   [scripting-container :- jruby-schemas/ConfigurableJRuby
    config :- jruby-schemas/JRubyConfig]
-  (.setEnvironment scripting-container (managed-environment (get-system-env) (:gem-home config)))
+  (.setEnvironment scripting-container (managed-environment (get-system-env) (:gem-home config) config))
```

Comment:
Just a style thing, but I might either:

- just pass in `config` and destructure the `:gem-home` and `:environment-vars` from within the `managed-environment` function 

or 

- destructure the config at this level and pass in the discrete values, like:

~~~clj
[{:keys [gem-home environment-vars]} :- jruby-schemas/JRubyConfig]
  (.setEnvironment scripting-container (managed-environment (get-system-env) gem-home environment-vars)
~~~

Pulling out one parameter but passing the whole config again as a second argument looks a little inconsistent to me.

Again, though, this is purely a cosmetic comment - no big deal.